examples/03_Batching/inference-batcher.cc

```diff
@@ -0,0 +1,398 @@
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <chrono>
+#include <thread>
+
+#include "YAIS/YAIS.h"
+#include "moodycamel/blockingconcurrentqueue.h"
+
+using yais::Context;
+using yais::Executor;
+using yais::Server;
+using yais::ThreadPool;
+
+using moodycamel::BlockingConcurrentQueue;
+using moodycamel::ConsumerToken;
+using moodycamel::ProducerToken;
+
+#include "simple.pb.h"
+#include "simple.grpc.pb.h"
+
+
+/**
+ * @brief Batching Service for Unary Requests
+ * 
+ * Exposes a Unary (send/recv) interface for a given RPC, but rather than
+ * computing the RPC, the service simply batches the incoming requests and
+ * forwards them via a gRPC stream to a service that implements the actual
+ * compute portion of the RPC.
+ * 
+ * The backend compute service is not a Unary service.  Rather it must
+ * implemented the LifeCycleBatching service Context, i.e. BatchingContext.
+ * The other application in this folder implements the backend service.
+ * 
+ * Streams are used as a forwarding mechanism because of how they interact
+ * with a load-balancer.  Unlike unary requests which get balanced on each
+ * request, a stream only get balanced when it is opened.  All items of a stream
+ * go to the same endpoint service.
+ * 
+ * 
+ * 
+ * @tparam ServiceType 
+ * @tparam Request 
+ * @tparam Response 
+ */
+template <class ServiceType, class Request, class Response>
+struct BatchingService
+{
+    using Callback = std::function<void(bool)>;
+
+    struct MessageType
+    {
+        Request *request;
+        Response *response;
+        Callback callback;
+    };
+
+    /**
+     * @brief Forwards incoming Unary requests via a gRPC Stream to
+     * a Batched Steaming Service that implements the actual RPC
+     */
+    class Client
+    {
+      public:
+        Client(
+            std::shared_ptr<::grpc::Channel> channel,
+            std::shared_ptr<ThreadPool> thread_pool)
+            : m_Stub(ServiceType::NewStub(channel)), m_ThreadPool(thread_pool), m_CurrentCQ(0)
+        {
+            for (int i = 0; i < m_ThreadPool->Size(); i++)
+            {
+                LOG(INFO) << "Starting Client Progress Engine #" << i;
+                m_CQs.emplace_back(new ::grpc::CompletionQueue);
+                auto cq = m_CQs.back().get();
+                m_ThreadPool->enqueue([this, cq] {
+                    ProgressEngine(*cq);
+                });
+            }
+        }
+
+        void WriteAndCloseStream(uint32_t messages_count, MessageType *messages)
+        {
+            auto cq = m_CQs[++m_CurrentCQ % m_CQs.size()].get();
+            LOG(INFO) << "Client using CQ: " << (void *)cq;
+
+            auto ctx = new Call;
+            for (int i = 0; i < messages_count; i++)
```

Comment:
`message_count` is a `uint32_t`
```cpp
for (uint32_t i = 0; i < messages_count; i++)
```

Reply:
Good catches.  I'm not getting these compiler warnings.  Are you compiling in the container, or using the VERBOSE flag to get the warnings?